arangod/VocBase/Methods/Queries.cpp

```diff
@@ -60,16 +64,77 @@ network::Headers buildHeaders() {
   return headers;
 }
 
-/// @brief return the list of currently running queries
-void getQueries(TRI_vocbase_t& vocbase, std::vector<aql::QueryEntryCopy> const& queries,
-                velocypack::Builder& out, char const* action, bool fanout) {
-  out.openArray();
+arangodb::Result checkAuthorization(TRI_vocbase_t& vocbase, bool allDatabases) {
```

Comment:
hm, can we find a less generic name for this function? and should it be static?

Reply:
It does not need to be static, because it is already in the anonymous namespace of the TU.
That will not make it visible to anything else but the translation unit, so it is the same as static linkage.

Reply:
It does seem to be specifically authorizing for access to the query list, so I guess it could be `checkAuthorizationForQueryList` if you really wanted to be more specific. But I think given the context it's not really necessary.